modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -350,6 +351,33 @@ public void testMergeKeysConflict() {
         assertEquals(Sql.DUPLICATE_KEYS_ERR, ex.code());
     }
 
+    /**
+     * Test verifies that scan is executed within provided transaction.
+     */
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-15081")
+    public void scanExecutedWithinGivenTransaction() {
+        sql("CREATE TABLE test (id int primary key, val int)");
+
+        Transaction tx = CLUSTER_NODES.get(0).transactions().begin();
+
+        sql(tx, "INSERT INTO test VALUES (0, 0)");
+
+        // just inserted row should be visible within the same transaction
+        assertEquals(1, sql(tx, "select * from test").size());
+
+        Transaction anotherTx = CLUSTER_NODES.get(0).transactions().begin();
```

Comment:
1. plz wrap CLUSTER_NODES.get(0).transactions() into different func, as it made for sql()
2. seems it helpful to check all nodes in cluster ? That all nodes not detect uncommited changes ?

Reply:
> plz wrap CLUSTER_NODES.get(0).transactions() into different func

ok, I'll do it next time, because this patch has already been merged

> seems it helpful to check all nodes in cluster ?

why do you think this will be helpful in context of current ticket?